lib/seahorn/BvOpSem2TrackingRawMemMgr.cc

```diff
@@ -74,36 +102,41 @@ TrackingRawMemManager::MemValTy TrackingRawMemManager::MemFill(
     TrackingRawMemManager::PtrTy dPtr, char *sPtr, unsigned int len,
     TrackingRawMemManager::MemValTy mem, uint32_t align) {
   RawMemValTy rawVal = m_main.MemFill(dPtr, sPtr, len, mem.getRaw(), align);
-  return MemValTy(rawVal, mem.getMetadata());
+  return MemValTy(rawVal, mem.getMetadata(READ), mem.getMetadata(WRITE),
+                  mem.getMetadata(ALLOC));
 }
 TrackingRawMemManager::MemValTy TrackingRawMemManager::MemCpy(
     TrackingRawMemManager::PtrTy dPtr, TrackingRawMemManager::PtrTy sPtr,
     Expr len, TrackingRawMemManager::MemValTy memTrsfrRead,
     TrackingRawMemManager::MemValTy memRead, uint32_t align) {
   RawMemValTy rawVal = m_main.MemCpy(dPtr, sPtr, len, memTrsfrRead.getRaw(),
                                      memRead.getRaw(), align);
-  return MemValTy(rawVal, memRead.getMetadata());
+  return MemValTy(rawVal, memRead.getMetadata(READ), memRead.getMetadata(WRITE),
+                  memRead.getMetadata(ALLOC));
 }
 TrackingRawMemManager::MemValTy TrackingRawMemManager::MemCpy(
     TrackingRawMemManager::PtrTy dPtr, TrackingRawMemManager::PtrTy sPtr,
     unsigned int len, TrackingRawMemManager::MemValTy memTrsfrRead,
     TrackingRawMemManager::MemValTy memRead, uint32_t align) {
   RawMemValTy rawVal = m_main.MemCpy(dPtr, sPtr, len, memTrsfrRead.getRaw(),
                                      memRead.getRaw(), align);
-  return MemValTy(rawVal, memRead.getMetadata());
+  return MemValTy(rawVal, memRead.getMetadata(READ), memRead.getMetadata(WRITE),
+                  memRead.getMetadata(ALLOC));
 }
 TrackingRawMemManager::MemValTy
 TrackingRawMemManager::MemSet(TrackingRawMemManager::PtrTy ptr, Expr _val,
                               Expr len, TrackingRawMemManager::MemValTy mem,
                               uint32_t align) {
   RawMemValTy rawVal = m_main.MemSet(ptr, _val, len, mem.getRaw(), align);
-  return MemValTy(rawVal, mem.getMetadata());
+  return MemValTy(rawVal, mem.getMetadata(READ), mem.getMetadata(WRITE),
+                  mem.getMetadata(ALLOC));
 }
 TrackingRawMemManager::MemValTy TrackingRawMemManager::MemSet(
     TrackingRawMemManager::PtrTy ptr, Expr _val, unsigned int len,
     TrackingRawMemManager::MemValTy mem, uint32_t align) {
   RawMemValTy rawVal = m_main.MemSet(ptr, _val, len, mem.getRaw(), align);
-  return MemValTy(rawVal, mem.getMetadata());
+  return MemValTy(rawVal, mem.getMetadata(READ), mem.getMetadata(WRITE),
+                  mem.getMetadata(ALLOC));
 }
 TrackingRawMemManager::MemValTy TrackingRawMemManager::storeValueToMem(
```

Comment:
this function should be factored out into the mixin. It simply dispatches into appropriate store function based on type of value.